vizro-core/examples/default/app_dev.py

```diff
@@ -41,8 +40,12 @@ def create_benchmark_analysis():
                 title="Click on a cell in country column:",
                 figure=dash_ag_grid(
                     data_frame=df,
+                    className="ag-theme-alpine vizro",
+                    # className="ag-theme-custom-theme",
+                    defaultColDef={"resizable": True, "sortable": True},
+                    columnSize="sizeToFit",
```

Comment:
I've noticed some strange behaviour but not sure if it's the configuration or some outer container we add on our side, but it seems like the ag-grid doesn't resize properly in the Vizro app?

E.g. if I start of with a smaller viewport and then just extend my screen, the app looks like this:

<img width="2559" alt="Screenshot 2024-01-17 at 16 27 51" src="https://github.com/mckinsey/vizro/assets/90609403/aee5d9fc-246d-49de-b1b5-34a4e42158b5">

After refreshing the page, it then resizes. Would be good to understand where this comes from and if we can fix it. Try to plug it into a pure dash app, if the issue still remains, then I think it's probably some outer container we add?

Reply:
@maxschulz-COL - found it! 
If you change this to  `columnSize="responsiveSizeToFit"`, it should fit to the entire width.

https://dash.plotly.com/dash-ag-grid/column-sizing#size-to-fit-and-responsive-size-to-fit

Reply:
The `sizeToFit` prop is working as intended.  It's called once when the grid is created.  The `responsiiveSizeToFit` will adjust the columns as the screen size changes , but users often find that annoying because they will lose any of the column widths that they set themselves.  

The [Column flex](https://dash.plotly.com/dash-ag-grid/column-sizing#column-flex) might be a better default.

Reply:
This is a tricky one. So before even going into this one, there are a few other questions of what we would like our default to be?

Do we want to our default behaviour even be size to fit? @huong-li-nguyen  The result look quite bad for big tables, but maybe there is a design guidance from Dan?:

![image](https://github.com/mckinsey/vizro/assets/83698606/aca6e6c7-073e-41de-8df7-3e4807c1016d)

If yes, I think I agree with @AnnMarieW , because the resizing can be quite annoying. I implemented the column flex as a default, I think it works well, and sizes to fit, but keeps proportions on resizing.

So the current proposed solution would be:
- size to fit space (let's discuss if this is desired)
- keep resized proportions upon window size adjustment with column flex
- user can of course overwrite this default to their liking


Wdyt?

Reply:
Flex can work well.  I usually like to add a min width with size to fit.  But whatever default is chosen, it's good that it's easy to overwrite :slightly_smiling_face:

Reply:
That is a good idea, might be the best of both worlds to some extent

Reply:
@AnnMarieW I have implemented the `flex` together with `minWidth` however it does not seem to extend the grid on large screens. Do you know how this could be done?

Li's suggestion (`responsiveSizeToFit`) fixes it in horizontal direction, but not in vertical - but we do not want to use it anyway due to the resizing

Reply:
@maxschulz-COL Try putting the` "flex": 1, "minWidth": 70` in the `defaultColDef` instead of the `columnDefs`

Reply:
This does the trick, thanks for the suggestion ❤️ 

Do you know why this is? Just curious why setting it in the defaults changes how this appears on the page?

Is this due to when the grid actually calculates its column widths?

Reply:
I'm not sure why it needs to be in the `defaultColDefs`, but you made a good guess.

Reply:
I think the current behaviour works nicely! Just one question @maxschulz-COL - in the example above you added as a screenshot, I would have expected a horizontal scrollbar to appear automatically. Is this happening when there are too many columns? In general, I think the expected behaviour should be:

- Columns should optimally always fit their content
- If there are too many columns and their contents gets cut off -> horizontal scrollbar should appear
- When a user re-sizes the window, the Table should resize as well

Reply:
Answered in this [comment](https://github.com/mckinsey/vizro/pull/268#discussion_r1504288071)